src/FirebaseProjectManager.php

```diff
@@ -64,29 +65,33 @@ protected function configure(string $name): FirebaseProject
 
         $config = $this->configuration($name);
 
-        if ($tenantId = $config['auth']['tenant_id'] ?? null) {
+        if ($tenantId = Arr::get($config, 'auth.tenant_id')) {
             $factory = $factory->withTenantId($tenantId);
         }
 
-        if ($credentials = $config['credentials']['file'] ?? null) {
-            $resolvedCredentials = $this->resolveCredentials((string) $credentials);
+        if ($credentials = Arr::get($config, 'credentials.file', Arr::get($config, 'credentials'))) {
+            if (is_string($credentials)) {
+                $factory = $factory->withServiceAccount($this->resolveCredentials($credentials));
+            }
 
-            $factory = $factory->withServiceAccount($resolvedCredentials);
+            if (is_array($credentials) && Arr::has($credentials, ['type', 'project_id'])) {
+                $factory = $factory->withServiceAccount($credentials);
+            }
         }
```

Comment:
As said above, I would prefer the introduction of `Arr` in another PR.

The `resolveCredentials()` method already does some of the checks, I would move the new functionality into `resolveCredentials()` and just™ do a

```php
if ($credentials = $config['credentials'] ?? null) {
    $resolvedCredentials = $this->resolveCredentials($credentials);
    $factory = $factory->withServiceAccount($credentials);
}
```

here.

Reply:
I think this must have been a comment of mine before I saw it, and that I edited afterwards.

Thank you for bearing with me and all the change requests! 🙏🏻